src/plugins/assume-nonnull.ts

```diff
@@ -0,0 +1,146 @@
+/**
+ * Copyright (c) 2016-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+import AlgebraicType = require('../algebraic-type');
+import Code = require('../code');
+import Error = require('../error');
+import FileWriter = require('../file-writer');
+import Maybe = require('../maybe');
+import ObjC = require('../objc');
+import ValueObject = require('../value-object');
+
+function assumeNonnullFileTransformation(request:FileWriter.Request):FileWriter.Request {
```

Comment:
We shouldn't be using a file transformation for this. It's going to make the plugin more brittle and make it more likely that it won't be compatible with other plugins which don't output things in the same way.

This is going to be more work, but I would recommend

- Adding a `nullability` property to `Class` in `objc.ts`
- It should be an enum type `ObjC.ClassNullability` that has something like `default` and `assumeNonnull`
- You can then add a function to the value and algebraic type plugins which can return a Maybe.Maybe<ObjC.ClassNullability>
- Your plugin can just `Maybe.Just(ObjC.ClassNullability.assumeNonnull`
- Change `objc-renderer.ts` to output the nullability macros for classes with the `assumeNonnull` value for `nullability`